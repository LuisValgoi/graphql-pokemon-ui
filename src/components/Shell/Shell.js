import React, { useState, useEffect } from 'react';
import { Redirect } from "react-router-dom";
import { useRouteMatch } from 'react-router-dom';

import BrowserURL from '../../util/BrowserURL';
import Container from 'react-bootstrap/Container';

const style = {
  shell: {
    backgroundColor: '#f7f7f7',
    borderBottom: '1px solid #ececec',
    position: 'fixed',
    width: '100%',
    zIndex: 100
  },
  breadcrumb: {
    position: 'absolute',
    marginTop: '18px'
  }
};

const Shell = ({ title }) => {
  const [showNavBackLink, setShowNavBackLink] = useState(false);

  useEffect(() => {
    setShowNavBackLink(window.location.pathname !== BrowserURL.HOME)
  }, [window.location.pathname]);

  return (
    <div style={style.shell} className='p-4'>
      {showNavBackLink && (
        <Container>
          <Redirect to={BrowserURL.HOME}>
            <p style={style.breadcrumb}>Voltar</p>
          </Redirect>
        </Container>
      )}
      <h1 className='text-center'>
        {title}
      </h1>
    </div>
  );
}

export default Shell;
