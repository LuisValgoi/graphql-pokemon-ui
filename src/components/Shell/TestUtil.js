import React from 'react'
import { render as rtlRender } from '@testing-library/react'
import { Provider } from 'react-redux'
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history'
import { createStore } from 'redux';
import BrowserURL from '../../util/BrowserURL';

const render = (ui, { mockStore, reducer, route = BrowserURL.LIST, ...renderOptions } = {}) => {
  const store = createStore(reducer, mockStore);
  const Wrapper = ({ children }) => {
    const history = createMemoryHistory({ initialEntries: [route] });
    return (
      <Provider store={store}>
        <Router history={history}>
          {children}
        </Router>
      </Provider>
    );
  };

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react'

export { render }
