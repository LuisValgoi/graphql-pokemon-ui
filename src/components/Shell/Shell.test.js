import React from 'react';

import { render, screen } from './TestUtil';
import '@testing-library/jest-dom/extend-expect';
import Shell from './Shell';
import PokemonDetailReducer from '../../redux/Pokemon/reducers/Detail';

let INITIAL_STATE = {};

beforeEach(() => {
  INITIAL_STATE = { pokemon: { detail: { item: { data: { id: 'UG9rZW1vbjowMDM=' } } } } };
  render(<Shell title='title' />, { reducer: PokemonDetailReducer, mockStore: INITIAL_STATE });
});

test('should render', () => {
  const shell = screen.getByTestId('shell-wrapper');

  expect(shell).toMatchSnapshot();
});

test('should have a little padding', () => {
  const shell = screen.getByTestId('shell-wrapper');

  expect(shell).toBeInTheDocument();
  expect(shell).toHaveClass('p-4');
});

test('home link should not be shown at home', () => {
  const shell = screen.getByTestId('shell-wrapper');
  const homeCol = screen.getByTestId('shell-home-col');
  const homeLink = screen.queryByTestId('shell-home-link');

  expect(shell).toBeInTheDocument();
  expect(shell).toContainElement(homeCol);
  expect(homeLink).toEqual(null);
});

test('home link should be shown at details page', () => {
  render(<Shell title='title' />, { reducer: PokemonDetailReducer, mockStore: INITIAL_STATE, route: '/pokemon/UG9rZW1vbjowMDM=' });

  const homeCol = screen.queryAllByTestId('shell-home-col')[1];
  const homeLink = screen.getByTestId('shell-home-link');

  expect(homeCol).toBeInTheDocument();
  expect(homeLink).toBeInTheDocument();
  expect(homeCol).toContainElement(homeLink);
  expect(homeLink).toHaveTextContent('Home');
});

test('edit button should not be shown at home', () => {
  const shell = screen.getByTestId('shell-wrapper');
  const editCol = screen.getByTestId('shell-edit-col');
  const editButton = screen.queryByTestId('shell-edit-button');

  expect(shell).toBeInTheDocument();
  expect(shell).toContainElement(editCol);
  expect(editCol).not.toContainElement(editButton);
});


test('edit button should be shown at detail', () => {
  render(<Shell title='title' />, { reducer: PokemonDetailReducer, mockStore: INITIAL_STATE, route: '/pokemon/UG9rZW1vbjowMDM=' });
  const editCol = screen.queryAllByTestId('shell-edit-col')[1];
  const editButton = screen.getByTestId('shell-edit-button');

  expect(editCol).toBeInTheDocument();
  expect(editButton).toBeInTheDocument();
  expect(editCol).toContainElement(editButton);
  expect(editButton).toHaveTextContent('Edit');
});
