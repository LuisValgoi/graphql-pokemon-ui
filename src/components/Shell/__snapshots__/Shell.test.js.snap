// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  class="p-4"
  data-testid="shell-wrapper"
  style="background-color: rgb(247, 247, 247); border-bottom: 1px solid #ececec; position: fixed; width: 100%; z-index: 100;"
>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="justify-content-start col-lg-1 col-md-1 col-sm-2 col-2"
        data-testid="shell-home-col"
      />
      <div
        class="col-lg-10 col-md-10 col-sm-8 col-8"
      >
        <h2
          class="text-center"
        >
          title
        </h2>
      </div>
      <div
        class="col-lg-1 col-md-1 col-sm-2 col-2"
        data-testid="shell-edit-col"
      />
    </div>
  </div>
</div>
`;
